src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests.cs

```diff
@@ -890,6 +891,42 @@ class C { }
             AssertEx.Equal(results.Diagnostics, fullDiagnostics);
         }
 
+        [Fact]
+        public void Cancellation_During_Initialization_Doesnt_Report_As_Generator_Error()
+        {
+            var source = @"
+class C 
+{
+}
+";
+            var parseOptions = TestOptions.Regular;
+            Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            compilation.VerifyDiagnostics();
+
+            Assert.Single(compilation.SyntaxTrees);
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+
+            var testGenerator = new CallbackGenerator(
+                onInit: (i) => { cts.Cancel(); },
+                onExecute: (e) => { }
+                );
+
+            // test generator cancels the token. Check that the call to this generator doesn't make it look like it errored.
+            var testGenerator2 = new CallbackGenerator2(
+                onInit: (i) => { i.CancellationToken.ThrowIfCancellationRequested(); },
+                onExecute: (e) => { throw ExceptionUtilities.Unreachable; });
+
+
+            GeneratorDriver driver = CSharpGeneratorDriver.Create(new[] { testGenerator, testGenerator2 }, parseOptions: parseOptions);
+            var oldDriver = driver;
+
+            Assert.Throws<OperationCanceledException>(() =>
+               driver = driver.RunGeneratorsAndUpdateCompilation(compilation, out var outputCompilation, out var outputDiagnostics, cts.Token)
+               );
+            Assert.Same(oldDriver, driver);
```

Comment:
This assert doesn't make sense to me -- if the RunGeneratorsAndUpdateCompilation throws (per the assertion) the assignment to driver won't ever happen.

Reply:
(this question may go to @chsienki since it looks like the pattern is being followed elsewhere, but the pattern doesn't make sense.)

Reply:
@chsienki thoughts on this comment? Should the Assert.Same just be removed in this case?

Reply:
Yeah, that assert isn't really doing anything. If its used elsewhere I suspect its because it previous was a try/catch then assert it wasn't assigned, vs. and actual assert.throws.